client/fidelity/compliance-review/controllers/ComplianceReviewController.js

```diff
@@ -1,32 +1,72 @@
-import {get} from 'lodash'
-import {getDateFilters} from 'superdesk-core/scripts/apps/search/directives/DateFilters'
-import CompliantLifetimeComponent from '../components/CompliantLifetime'
-import VersionCreatedComponent from '../components/VersionCreated'
-import {getStatus} from '../compliance-status'
-
-ComplianceReviewCtrl.$inject = ['$location', 'config', 'moment'];
-export function ComplianceReviewCtrl($location, config, moment) {
-    const VIEW_DATE_FORMAT = config.view.dateformat;
+import { get } from 'lodash';
+import moment from 'moment';
+import { getDateFilters } from 'superdesk-core/scripts/apps/search/directives/DateFilters';
+import CompliantLifetimeComponent from '../components/CompliantLifetime';
+import VersionCreatedComponent from '../components/VersionCreated';
+import { getStatus } from '../compliance-status';
+
+const filterToDaysDiff = (filter) => {
+    switch (filter) {
+        case 'before_next_month': return 30;
+        case 'before_3_months_ahead': return 90;
+        default: return 0;
+    }
+};
+
+// old versions of corrected items can
+// have a date that doesn't match the filter
+const filterWrongLifetime = (items, filter) => {
+    const range = filterToDaysDiff(filter);
+    const now = moment();
+    return items.filter(({ archive_item }) => {
+        const lifetime = moment(archive_item.extra.compliantlifetime);
+        return lifetime.diff(now, 'days') < range;
+    });
+};
+
+const unwantedStates = [ 'killed' ];
+const filterUnwatedStates = (items) =>
+    items.filter(item => !unwantedStates.includes(item.state));
+const wantedItemTypes = [ 'text' ];
+const filterUnwatedTypes = (items) =>
+    items.filter(item => wantedItemTypes.includes(item.type));
+
+ComplianceReviewCtrl.$inject = ['$location', 'moment', 'gettext', '$scope'];
+export function ComplianceReviewCtrl($location, moment, gettext, $scope) {
     const SUPERDESK = 'local';
 
-    const compliantFilter = getDateFilters(gettext)
-        .find(f => f.fieldname === 'extra.compliantlifetime');
+    const watchItems = (items) => {
+        if (items) {
+            let {_items} = items
+            _items = filterWrongLifetime(_items, getFilterFromUrl());
+            _items = filterUnwatedStates(_items);
```

Comment:
Move fetching of items to the server side. Query items you want instead of querying everything and excluding what you don't want.

Reply:
@tomaskikutis that was my first approach but I can't figure out how to make a custom query because right now I'm using the custom `getSearch` of client-core and that only admits specific filters so I think that would require modifying search results again

Reply:
@tomaskikutis any suggestion?